crates/miden-lib/src/testing/account_component/mock_account_component.rs

```diff
@@ -22,25 +22,29 @@ impl MockAccountComponent {
     // --------------------------------------------------------------------------------------------
 
     /// Constructs a [`MockAccountComponent`] with empty storage.
-    pub fn with_empty_slots() -> Result<Self, AccountError> {
+    pub fn with_empty_slots() -> Self {
         Self::new(vec![])
     }
 
     /// Constructs a [`MockAccountComponent`] with the provided storage slots.
-    pub fn with_slots(storage_slots: Vec<StorageSlot>) -> Result<Self, AccountError> {
+    ///
+    /// # Panics
+    ///
+    /// Panics if the number of slots exceeds the maximum.
+    pub fn with_slots(storage_slots: Vec<StorageSlot>) -> Self {
         Self::new(storage_slots)
     }
 
     // HELPERS
     // --------------------------------------------------------------------------------------------
 
-    fn new(storage_slots: Vec<StorageSlot>) -> Result<Self, AccountError> {
+    fn new(storage_slots: Vec<StorageSlot>) -> Self {
```

Comment:
I'm not sure about this one, but maybe we could also check against `AccountStorage::MAX_NUM_STORAGE_SLOTS` here, instead of trying from `u8`?